File: src/org/onion_lang/onion/compiler/env/StaticImportList.java
/* ************************************************************** *
 *                                                                *
 * Copyright (c) 2005, Kota Mizushima, All rights reserved.       *
 *                                                                *
 *                                                                *
 * This software is distributed under the modified BSD License.   *
 * ************************************************************** */
package org.onion_lang.onion.compiler.env;

import java.util.ArrayList;
import java.util.List;

/**
 * A type safe import list.
 * @author Kota Mizushima
 * 
 */
public class StaticImportList {
  private List items = new ArrayList();

  public StaticImportList(){
  }
  
  public void add(StaticImportItem item){
    items.add(item);
  }
  
  public String get(int index){
    return (String)items.get(index);
  }
  
  public StaticImportItem[] getItems(){
    return (StaticImportItem[])items.toArray(new StaticImportItem[0]);
  }
  
  public int size(){
    return items.size();
  }
}

File: src/main/scala/onion/compiler/StaticImportList.scala
/* ************************************************************** *
 *                                                                *
 * Copyright (c) 2005-2012, Kota Mizushima, All rights reserved.       *
 *                                                                *
 *                                                                *
 * This software is distributed under the modified BSD License.   *
 * ************************************************************** */
package onion.compiler

import java.util.ArrayList
import java.util.List
import collection.mutable.ArrayBuffer

/**
 * A type safe import list.
 * @author Kota Mizushima
 *
 */
class StaticImportList {
  private val items = new ArrayBuffer[StaticImportItem]

  def add(item: StaticImportItem): Unit =  items += item

  def get(index: Int): StaticImportItem = items(index)

  def getItems: Array[StaticImportItem] = items.toArray

  def size: Int = items.size
}